packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -152,6 +153,14 @@ export async function fetchServerResponse(
       return doMpaNavigation(responseUrl.toString())
     }
 
+    // We may navigate to a page that requires a different Webpack runtime.
+    // In prod, every page will have the same Webpack runtime.
+    // In dev, the Webpack runtime is minimal for each page.
+    // We need to ensure the Webpack runtime is updated before executing client-side JS of the new page.
+    if (process.env.NODE_ENV !== 'production') {
+      await waitForWebpackRuntimeHotUpdate()
+    }
```

Comment:
I guess we can safely ignore prefetches here?

Reply:
right, in dev this function won't see any prefetches

Reply:
Which wording are you referring to?

Reply:
"prefetchKind" being defined even though we're not prefetching. In my head I'm mixing this in with prefetches in the browser e.g. `<Link prefetch />` but it seems these are different concepts?

Reply:
Yeah I agree it's a bit confusing.  Part of what's confusing is that prefetches are tightly coupled to navigations. A navigation cannot exist without a prefetch (it's what the reducer acts upon when changing the router state), so that's why we have the concept of an "temporary" prefetch.

I don't think you're mixing the concepts though, `<Link prefetch />` does indeed _mostly_ map to what `prefetchKind` is, with the exception of the temporary one as you've noted. The comments in this file could use updating.

- `PrefetchKind.FULL` = `prefetch={true}`
- `PrefetchKind.AUTO` = prefetch left unspecified, or `prefetch={null}`
- `PrefetchKind.TEMPORARY` = A placeholder for a navigation that occurred without a regular prefetch. For example if you have `prefetch={false}`, or you're in dev, a temporary prefetch is created. If we later discover a prefetch intent, `TEMPORARY` will get replaced with whatever that intent is.

Reply:
> I haven't looked into 'temporary' vs undefined more. 

The `undefined` case is any other time we need to retrieve RSC data from the server when not directly associated with the navigation event. These come to mind but might not be exhaustive:

- "Lazy fetching" missing segments in layout-router (this is pre-PPR behavior)
- The dynamic data request in PPR (analogous to the lazy fetch above, but doesn't happen during render)
- Refreshing segment data as a result of something like `router.refresh()`